integration/test/performancetest/transmitter.go

```diff
@@ -153,12 +197,194 @@ func (transmitter *TransmitterAPI) TableExist() (bool, error) {
+Params:
+	hash: this is the commitHash
+	targetAttributes: this is the targetAttribute to be added to the dynamo item
+	testHash: this is the hash of the last item, used like a version check
+*/
+func (transmitter * TransmitterAPI) UpdateItem(hash string,targetAttributes map[string]types.AttributeValue) error{
+	var err error
+	var ae *types.ConditionalCheckFailedException // this exception represent the the atomic check has failed
+	fmt.Println("Updating:",hash)
+	item,err := transmitter.Query(hash) // get most Up to date item from dynamo | O(1) bcs of global sec. idx.
+	if len(item) ==0{ // check if hash is in dynamo
+		return errors.New("ERROR: Hash is not found in dynamo")
+	}
+	commitDate := fmt.Sprintf("%d",int(item[0]["CommitDate"].(float64)))
+	year := fmt.Sprintf("%d",int(item[0]["Year"].(float64)))
+	testHash := item[0][TEST_HASH].(string)
+	//setup the update expression
+	expressionAttributeValues := make(map[string]types.AttributeValue)
+	expression := ""
+	n_expression := len(targetAttributes)
+	i :=0
+	for attribute, value := range targetAttributes{
+		expressionName := ":" +strings.ToLower(attribute)
+		expression = fmt.Sprintf("set %s = %s",attribute,expressionName)
+		expressionAttributeValues[expressionName] = value
+		if(n_expression -1 >i){
+			expression += "and"
+		}
+		i++
+	}
+	expressionAttributeValues[":testHash"] = &types.AttributeValueMemberS{Value: testHash}
+	//----
+	//call update
+	_, err = transmitter.dynamoDbClient.UpdateItem(context.TODO(), &dynamodb.UpdateItemInput{
+        TableName: aws.String(transmitter.DataBaseName),
+        Key: map[string]types.AttributeValue{
+            "Year": &types.AttributeValueMemberN{Value: year},
+			"CommitDate": &types.AttributeValueMemberN{Value: commitDate },
+        },
+        UpdateExpression: aws.String(expression),
+        ExpressionAttributeValues: expressionAttributeValues,
+		ConditionExpression: aws.String("#testHash = :testHash"),
+		ExpressionAttributeNames: map[string]string{
+			"#testHash": TEST_HASH,
+		},
+    })
 
+    if err != nil && !errors.As(err,&ae) {
+        panic(err)
+    }
+	return err
+}
+/*
+UpdateReleaseTag()
+Desc: This function takes in a commit hash and updates the release value to true 
+Param: commit hash in terms of string 
+*/
+func (transmitter * TransmitterAPI) UpdateReleaseTag(hash string) error{
+	var err error
+	var ae *types.ConditionalCheckFailedException
+	attributes := map[string]types.AttributeValue{
+		"isRelease":&types.AttributeValueMemberBOOL{Value: true},
+	}
+	for {//concurrency retry
+		/*solving parallel updates using an optimistic lock and retry,
+		this may result in a livelock;
+		however, random sleeps makes this possiblity nearly impossible*/
+		err = transmitter.UpdateItem(hash,attributes) //try to update the item
+		//this may be overwritten by other test threads, in that case it will return a specific error
+		if errors.As(err,&ae){ //check if our call got overwritten
```

Comment:
The comparison seems duplicate to [this code](https://github.com/okankoAMZ/amazon-cloudwatch-agent/blob/9676dd418451dbbfcb4edaf938a39293f6d818f2/integration/test/performancetest/transmitter.go#L333-L335) and other code. Therefore, would it be best for other function to return the error and the corresponding function who uses the function would handle how error instead? Since I favor more in centralized handling error

Reply:
I am only using in two places thats why I though it would be okay. However, if you want I can move it inside UpdateItem() directly.

Reply:
This is such a small bit of code there's no reason to make a separate function

Reply:
No the idea was too avoid using `errors.As` less as possible. Therefore, only return err and let's other function dealing with the specific error instead of panic the error here.

Reply:
The point of this code is to capture and handle an expected error. I'm not sure why it would be appropriate to return an error here.

Reply:
We don't need to populate `ae` with `errors.As`. We only need to verify that the conditional check failure happened. You should change the check to `errors.Is`

Reply:
Nvm. you still need `errors.As`

Reply:
Maybe its only my design so I leave the choice to you guys.